python/letsql/backends/let/tests/test_client.py

```diff
@@ -75,3 +77,30 @@ def test_register_table_with_uppercase_multiple_times(ls_con):
     assert uppercase_table_name in ls_con.list_tables()
     assert ls.execute(t) is not None
     assert t.schema() == expected_schema
+
+
+@pytest.mark.parametrize(
+    "keys",
+    [
+        param(tuple(), id="empty"),
+        param((ls.asc("yearID"),), id="one-column"),
+        param((ls.asc("yearID"), ls.desc("stint")), id="two-columns"),
+    ],
+)
+def test_register_record_batch_reader_sorted(batting_df, keys):
+    con = ls.connect()
+    t = con.register(batting_df, "batting")
+
+    if keys:
+        t = t.order_by(*keys)
+
+    record_batch_reader = t.select("yearID", "stint").to_pyarrow_batches()
+    t = con.register(
+        record_batch_reader,
+        "t2",
+        ordering=[key.resolve(t) for key in keys],
+    )
+    expr = t.group_by("yearID").agg(max_tiny=t["stint"].max())
+
+    physical_plan = ls.get_plans(expr)["physical_plan"]
+    assert not keys or "ordering_mode=Sorted" in physical_plan
```

Comment:
I think we should make it explicit that we expect `"ordering_mode=Sorted"` to **not** be in `physical_plan` when there is no ordering
```
    ordering_string = "ordering_mode=Sorted"
    if keys:
        assert ordering_string in physical_plan
    else:
        assert ordering_string not in physical_plan
```